File: frontend/src/types/schedule.ts
export type ScheduleStatus = "Draft" | "Active" | "Paused" | "Expired";
export type ScheduleRepeat = "Once" | "Daily" | "Weekdays" | "Weekends" | "Weekly" | "Monthly";
export type SchedulePriority = "Low" | "Normal" | "High" | "Emergency";

export const SCHEDULE_REPEATS: ScheduleRepeat[] = ["Once", "Daily", "Weekdays", "Weekends", "Weekly", "Monthly"];
export const SCHEDULE_PRIORITIES: SchedulePriority[] = ["Emergency", "High", "Normal", "Low"];

export interface Schedule {
  id: string;
  name: string;
  playlistId: string;
  deviceIds: string[];
  startDate: string; // YYYY-MM-DD
  endDate: string; // YYYY-MM-DD
  startTime: string; // HH:mm
  endTime: string; // HH:mm
  repeat: ScheduleRepeat;
  priority: SchedulePriority;
  status: ScheduleStatus;
  createdAt?: number;
  updatedAt?: number;
  /** Owning client; null or absent means it belongs to the operator. Only administrators can change it. */
  clientId?: string | null;
}

export type ScheduleCreatePayload = Omit<Schedule, "id" | "createdAt" | "updatedAt">;
export type ScheduleUpdatePayload = Partial<ScheduleCreatePayload>;

File: backend/app/schemas/schedule.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import date, time

class ScheduleBase(BaseModel):
    name: str
    playlistId: str
    startDate: date
    endDate: date
    startTime: time
    endTime: time
    repeat: str
    priority: str
    status: str

class ScheduleCreate(ScheduleBase):
    deviceIds: List[str]

class ScheduleUpdate(BaseModel):
    name: Optional[str] = None
    playlistId: Optional[str] = None
    startDate: Optional[date] = None
    endDate: Optional[date] = None
    startTime: Optional[time] = None
    endTime: Optional[time] = None
    repeat: Optional[str] = None
    priority: Optional[str] = None
    status: Optional[str] = None
    deviceIds: Optional[List[str]] = None

class ScheduleResponse(ScheduleBase):
    id: str
    deviceIds: List[str] = []
    createdAt: int
    updatedAt: int

    class Config:
        from_attributes = True
